kornia/augmentation/utils/helpers.py

```diff
@@ -82,6 +82,34 @@ def _infer_batch_shape3d(input: Union[Tensor, Tuple[Tensor, Tensor]]) -> torch.S
     return tensor.shape
 
 
+def _transform_input_by_shape(input: Tensor, reference_shape: torch.Size, match_channel: bool = True) -> Tensor:
+    """Reshape an input tensor to have the same dimensions as the reference_shape.
+
+    Arguments
+        input: tensor to be transformed
+        reference_shape: shape used as reference
+        match_channel: if True, C_{src} == C_{ref}. otherwise, no constrain. C =1 by default
+    """
+    B = reference_shape[-4] if len(reference_shape) >= 4 else None
+    C = reference_shape[-3] if len(reference_shape) >= 3 else None
+
+    if len(input.shape) == 2:
+        input = input.unsqueeze(0)
+
+    if len(input.shape) == 3 and B == input.shape[-3]:
+        # If the first dim matches within the batch_size, add a `C` dim
+        # Useful to handler Masks without `C` dimensions
+        input = input.unsqueeze(1)
+
```

Comment:
should the unsqueeze(0) be needed in an else here I think -- tests didn't cover it? 🤔

Reply:
It is fine if we only support HW, CHW, BHW. We did not think about BHW cases I think.

Reply:
Ah. I got your question. No, the logic is correct. If we have HW, it will be firstly transformed into CHW. Then BCHW